<launch>
  
  <arg name="output" default="screen"/>

  <node pkg="predicator_planning"
    name="predicator_planning_node"
    type="predicator_planning_node"
    output="$(arg output)">

    <rosparam param="description_list">
      - /gazebo/barrett_manager/robot_description
      - /gazebo/w2barrett_manager/robot_description
      - /peg1_description
      - /peg2_description
      - /ring1_description
      - /stage_model
    </rosparam>

    <rosparam param="joint_state_topic_list">
      - /gazebo/barrett_manager/joint_states
      - /gazebo/w2barrett_manager/joint_states
    </rosparam>

    <rosparam param="floating_root_list">
      - id: ring1
        frame: ring1/ring_link
    </rosparam>

    <param name="verbosity" value="2"/>
    <param name="world_frame" value="/world"/>

    <param name="height_threshold" value="0.1"/>
    <param name="rel_x_threshold" value="0.1"/>
    <param name="rel_y_threshold" value="0.1"/>
    <param name="rel_z_threshold" value="0.1"/>
    <param name="near_2D_threshold" value="0.2"/>
    <param name="near_3D_threshold" value="0.25"/>

    <rosparam param="frames">
      - ring1/ring_link
      - peg1/peg_link
      - peg1/base_link
      - peg1/peg_top_link
      - peg2/peg_link
      - peg2/base_link
      - peg2/peg_top_link
      - wam/shoulder_yaw_link
      - wam/shoulder_pitch_link
      - wam/upper_arm_link
      - wam/forearm_link
      - wam/wrist_yaw_link
      - wam/wrist_pitch_link
      - wam/wrist_palm_link
      - wam2/shoulder_yaw_link
      - wam2/shoulder_pitch_link
      - wam2/upper_arm_link
      - wam2/forearm_link
      - wam2/wrist_yaw_link
      - wam2/wrist_pitch_link
      - wam2/wrist_palm_link
      - stage_link
    </rosparam>


  </node>
</launch>
